R/reduce.R

```diff
@@ -491,6 +501,17 @@ accumulate_names <- function(nms, init, dir) {
   nms
 }
 
+acc_simplify <- function(x) {
+  # It would be unappropriate to simplify the result rowwise with
+  # `accumulate()` because it has invariants defined in terms of
+  # `length()` rather than `vec_size()`
+  if (some(x, is.data.frame)) {
+    x
+  } else {
+    vec_simplify(x)
```

Comment:
I think this is the only place that `vec_simplify()` is used, so it probably should be incorporated into this function.